<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
#[doc(keyword = "match")] //~ ERROR: `#[doc(keyword = "...")]` is experimental
=======
#[doc(keyword = "match")] //~ ERROR: `#[doc(keyword)]` is experimental
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
/// wonderful
mod foo{}

fn main() {}
