<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
#[doc(include="asdf.md")] //~ ERROR: `#[doc(include = "...")]` is experimental
=======
#[doc(include="asdf.md")] //~ ERROR: `#[doc(include)]` is experimental
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
fn main() {}
