<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
#[doc(cfg(unix))] //~ ERROR: `#[doc(cfg(...))]` is experimental
=======
#[doc(cfg(unix))] //~ ERROR: `#[doc(cfg)]` is experimental
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
fn main() {}
